src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -0,0 +1,173 @@
+        }
+    }
+
+    @KafkaListener(topics = ORDER_RECEIVED_TOPIC_RETRY, groupId = ORDER_RECEIVED_GROUP_RETRY,
+            autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
+    public void processOrderReceivedRetry(String orderReceivedUri)
+            throws SerializationException, ExecutionException, InterruptedException {
+        try {
+            logMessageReceived(orderReceivedUri, ORDER_RECEIVED_TOPIC_RETRY);
+        } catch (Exception x){
+            republishMessageToTopic(orderReceivedUri, ORDER_RECEIVED_TOPIC_RETRY, ORDER_RECEIVED_TOPIC_ERROR, x.getMessage());
+        }
+    }
+
+    @KafkaListener(topics = ORDER_RECEIVED_TOPIC_ERROR, groupId = ORDER_RECEIVED_GROUP_ERROR,
+            autoStartup = "${uk.gov.companieshouse.item-handler.error-consumer}")
+    public void processOrderReceivedError(String orderReceivedUri) {
+        try {
+            logMessageReceived(orderReceivedUri, ORDER_RECEIVED_TOPIC_ERROR);
+        } catch (Exception x){
+            LOGGER.error("Message: " + orderReceivedUri + " received on topic: " + ORDER_RECEIVED_TOPIC_ERROR + " could not be processed.");
+        }
+    }
+
+    private ConsumerConfig getConsumerConfig() {
+        ConsumerConfig consumerConfig = new ConsumerConfig();
+        consumerConfig.setMaxRetries(MAX_RETRY_ATTEMPTS);
+        consumerConfig.setKeyDeserializer(StringDeserializer.class.getName());
+        consumerConfig.setValueDeserializer(StringDeserializer.class.getName());
+        consumerConfig.setResetOffset(false);
+        consumerConfig.setBrokerAddresses(brokerAddresses.split(","));
+        consumerConfig.setAutoCommit(true);
+        return consumerConfig;
+    }
+
+    private ProducerConfig getProducerConfig() {
+        ProducerConfig config = new ProducerConfig();
+        if (brokerAddresses != null && !brokerAddresses.isEmpty()) {
+            config.setBrokerAddresses(brokerAddresses.split(","));
+        } else {
+            throw new ProducerConfigException("Broker addresses for kafka broker missing, check if environment variable KAFKA_BROKER_ADDR is configured. " +
+                    "[Hint: The property 'kafka.broker.addresses' uses the value of this environment variable in live environments " +
+                    "and that of 'spring.embedded.kafka.brokers' property in test.]");
+        }
+
+        config.setRoundRobinPartitioner(true);
+        config.setAcks(Acks.WAIT_FOR_ALL);
+        config.setRetries(10);
+        return config;
+    }
+
+    protected void republishMessageToTopic(String orderUri, String currentTopic, String nextTopic, String errorMessage)
+            throws SerializationException, ExecutionException, InterruptedException {
+        LOGGER.error("Processing message: " + orderUri + " received on topic: " + currentTopic
+                + " failed with exception: " + errorMessage);
```

Comment:
Good to see this logging. Do you think this will provide enough help to troubleshooters should some unexpected exception arise? It could be useful to see the exception type as well as the stack trace perhaps?

Reply:
Good point. I'll change this to include exception details.